src/metabase/query_processor/middleware/auto_bucket_datetimes.clj

```diff
@@ -71,6 +71,8 @@
         (not (every? auto-bucketable-value? vs)))))
    ;; do not auto-bucket fields inside a `:time-interval` filter -- it already supplies its own unit
    (mbql.u/is-clause? :time-interval x)
+   ;; do not auto-bucket fields inside a `:datetime-diff` clause: the precise timestamp is needed for the difference
+   (mbql.u/is-clause? :datetime-diff x)
```

Comment:
`is-clause?` can check multiple clauses at once like `#{:time-interval :datetime-diff}` so you could combine these into one function call if you wanted. Not that it makes a big difference but it is slightly more concise